src/renderer/components/inputs/SliderInput.tsx

```diff
@@ -52,22 +53,27 @@ const SliderInput = memo(
         ends,
         noteExpression,
         accentColor,
+        hideTrailingZeros,
         isLocked,
     }: SliderInputProps) => {
         const [input, setInput] = useInputData<number>(inputId);
         const [inputString, setInputString] = useState(String(input));
         const [sliderValue, setSliderValue] = useState(input ?? def);
         const [showTooltip, setShowTooltip] = useState(false);
 
+        const precision = Math.max(getPrecision(offset), getPrecision(step));
+        const precisionOutput = (val: number) =>
+            hideTrailingZeros ? String(val) : val.toFixed(precision);
+
         useEffect(() => {
             setSliderValue(input ?? def);
             if (!Number.isNaN(input)) {
-                setInputString(String(input));
+                setInputString(input === undefined ? String(input) : precisionOutput(input));
```

Comment:
I just thought it would be funny lol. /s

Clearly I would have fixed it if I knew I'd found a bug. What is it? That input can potentially be undefined? input/setInput has been like that the whole time, I think. I thought it was an intended part of the design that input could be undefined.

Reply:
`String(undefined) === "undefined"`, which isn't a nice thing to display to users. Just like the slider value, the input string should probably default to `def`.

Reply:
When I was sorting migrations the other day, I was getting undefined in one of the number boxes when I messed it up, so I think this is actually an issue in some existing numberInput code as well.

Reply:
Good catch. The `const [inputString, setInputString] = useState(String(input));` there should be `const [inputString, setInputString] = useState(String(input ?? def));` and the `if (!Number.isNaN(asNumber) && !areApproximatelyEqual(asNumber, input!))` there should have `!== undefined` check as well.